#!/usr/bin/env python
PACKAGE = "cheetah_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("ori_roll", double_t, 0, "", 0.25, 0, 12)
gen.add("ori_pitch", double_t, 0, "", 0.25, 0, 12)
gen.add("ori_yaw", double_t, 0, "", 10, 0, 12)
gen.add("pos_x", double_t, 0, "", 2, 0, 20)
gen.add("pos_y", double_t, 0, "", 2, 0, 20)
gen.add("pos_z", double_t, 0, "", 10, 0, 20)

gen.add("rate_roll", double_t, 0, "", 0, 0, 1)
gen.add("rate_pitch", double_t, 0, "", 0, 0, 1)
gen.add("rate_yaw", double_t, 0, "", 0.3, 0, 1)
gen.add("vel_x", double_t, 0, "", 0.2, 0, 1.5)
gen.add("vel_y", double_t, 0, "", 0.2, 0, 1.5)
gen.add("vel_z", double_t, 0, "", 0.6, 0, 1.5)

gen.add("alpha", double_t, 0, "Force weight", 1e-6, 0, 1e-4)

gen.add("horizon", int_t, 0, " ", 10, 0, 20)
gen.add("dt", double_t, 0, " ", 0.015, 0, 0.1)
gen.add("update_rate", double_t, 0, " ", 66.66, 20, 100)

exit(gen.generate(PACKAGE, "weight", "Weight"))
